apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +28,61 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
-        p.mkpath unless p.exist?
+      OodAppkit.dataroot.join('projects').tap do |path|
+        path.mkpath unless path.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /\A[\w-]+\z/,
+    message: 'Directory may only contain letters, digits, dashes, and underscores'
+  }
+
   attr_reader :dir
+  delegate :icon, :name, :description, to: :manifest
+
+  def initialize(attributes = {})
+    @dir            = attributes.fetch(:dir, nil).to_s
+  end
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def save
+    result = update( {} )
+    errors.add(:save, 'Cannot save manifest') unless result
+    result
   end
 
-  def config_dir
-    Pathname.new("#{absolute_dir}/.ondemand").tap { |p| p.mkpath unless p.exist? }
+  def update(attributes)
+    manifest = Manifest.load(manifest_path)
+    manifest = manifest.merge(attributes)
+    manifest.valid? ? manifest.save(manifest_path) : false
   end
```

Comment:
I don't know if I like reusing `manifest` as a variable name but I think if you swapped who's merged it may make initialization a little easier to reason about.

Manifest is a read-only object so really the only way to pass things from the controller down to are through params.

I'm going to have to dig into this further to see how we can get around updating a delegated thing  and a `form_for` situation. 

```suggestion
  def update(attributes)
    m = Manifest.new(attributes)
    m = m.merge(manifest.to_h)
    m.valid? ? m.save(manifest_path) : false
  end
```

Reply:
I like semantically meaningful names from a maintainability standpoint. Ultimately, we can see what is being checked `manifest.valid?` which is easy to read. Personally I don't like short, meaningless names as they introduce cognitive load/overhead, because i'd have to look at this a bit to understand `m.valid?` to understand what `m` is if I came back 6 months from now.

Do we even need the merge actually? The `Manifest` class will do that on initialization it looks like:
      
      # app/manifest.rb
      def initialize(opts)
        raise InvalidContentError.new unless(opts && opts.respond_to?(:to_h))

        @manifest_options = opts.to_h.with_indifferent_access
      end

Reply:
I agree with Travis on this. We really should get in the habit of using meaningful variable names.

Reply:
> Do we even need the merge actually? The Manifest class will do that on initialization it looks like:

Great question! Our manifest is read-only, so we don't need to actually here. You could just create something new and save it if it's valid because that's how we're treating that object. In an all or nothing fashion.